pkg/symbolizer/addr2line.go

```
@@ -163,7 +163,7 @@ func parse(interner *Interner, s *bufio.Scanner) ([]Frame, error) {
 		}
 		file := ln[:colon]
 		line, err := strconv.Atoi(ln[colon+1 : lineEnd])
-		if err != nil || fn == "" || fn == "??" || file == "" || file == "??" || line <= 0 {
+		if err != nil || fn == "" || fn == "??" || file == "" || file == "??" || line < 0 {
```

I understand your motivation now.
The question is how will the line 0 be interpreted by our UI.
Do you see any side effects locally?

---

My recommendation is to set Line to -1 in Frame{}, line 173.
It will allow to fail fast on misuses.

---

The alternative may be to find the line with function name and mark it as a coverage point.
But it looks too expensive...

---

> My recommendation is to set Line to -1 in Frame{}, line 173. It will allow to fail fast on misuses.

-1 will cause problem here https://github.com/google/syzkaller/blob/master/pkg/cover/report.go#L111.
```
		ln := f.lines[frame.StartLine]
```

---

It looks Ok. Can you skip the frames with line == -1 there?
We use the same structures to export information about line coverage. Exporting data for line 0 we'll confuse customers.

If we can't map frame to the line but can map frame to the function, let's do it.
My concern are the frames from multiple functions mapped to the line 0. They'll go to export which we don't want to happen.